
<launch>
  <!-- arguments for camera input -->
  <arg name="dictionary" value="8"/>
  <arg name="fiducial_len" value="0.120" />
  <arg name="image" default="image_raw"/>
  <arg name="transport" default="compressed"/>
  <arg name="do_pose_estimation" default="true"/>
  <arg name="publish_images" default="true"/>
  <arg name="publish_tf_transforms" default="true"/>
  <arg name="publish_fiducial_tf" default="false"/>
  <arg name="ignore_fiducials" default="" />
  <arg name="fiducial_len_override" default="" />
  <arg name="filter_size" default="0.005"/>
  <arg name="x_min" default="-5"/>
  <arg name="x_max" default="5"/>
  <arg name="y_min" default="-5"/>
  <arg name="y_max" default="5"/>
  <arg name="z_min" default="-5"/>
  <arg name="z_max" default="5"/>
  <!-- <arg name="target_fiducial_id" default="1"/> -->
  
  <arg name="camera_map" default="$(find assembly_camera_manager)/maps/camera"/>
  <arg name="world_map" default="$(find assembly_camera_manager)/maps/world/snu.yaml"/>

  <!-- Start camera broadcaster-->
  <node name="single_azure_manager" pkg="assembly_camera_manager" 
        type="azure_manager.py" respawn="false" output="screen" >
      <param name="camera_name" type="string" value="azure1" />
      <param name="camera_map" value="$(arg camera_map)" />
      <param name="world_map" value="$(arg world_map)" />
      <param name="filter_size" value="$(arg filter_size)" />
      <param name="x_min" value="$(arg x_min)" />
      <param name="x_max" value="$(arg x_max)" />
      <param name="y_min" value="$(arg y_min)" />
      <param name="y_max" value="$(arg y_max)" />
      <param name="z_min" value="$(arg z_min)" />
      <param name="z_max" value="$(arg z_max)" />

  </node>

  <!-- Start the aruco_node for azure-->
  <node pkg="aruco_detect" name="aruco_detect" ns="azure1" type="aruco_detect" respawn="false">
    <param name="image_transport" value="$(arg transport)"/>
    <param name="publish_images" value="$(arg publish_images)" />
    <param name="publish_tf_transforms" value="$(arg publish_tf_transforms)" />
    <param name="publish_fiducial_tf" value="$(arg publish_fiducial_tf)" />
    <param name="fiducial_len" value="$(arg fiducial_len)"/>
    <param name="dictionary" value="$(arg dictionary)"/>
    <param name="do_pose_estimation" value="$(arg do_pose_estimation)"/>
    <param name="ignore_fiducials" value="$(arg ignore_fiducials)"/>
    <param name="fiducial_len_override" value="$(arg fiducial_len_override)"/>
    <remap from="tf_prefix" to="azure1"/>
    <remap from="camera/compressed" to="/azure1/rgb/$(arg image)/$(arg transport)"/>
    <remap from="camera_info" to="/azure1/rgb/camera_info"/>
    <remap from="fiducial_images" to="/azure1/fiducial_images"/>
    <remap from="fiducial_transforms" to="/azure1/fiducial_transforms"/>
    <remap from="fiducial_vertices" to="/azure1/fiducial_vertices"/>
  </node>


  <!-- define camera2_map as a map  -->
  <node pkg="tf" type="static_transform_publisher" 
        name="azure_map_table_broadcaster" args="0 0 1 0 0 0 map map_table 100" /> 
  <node pkg="tf" type="static_transform_publisher" 
        name="azure_base_broadcaster" args="0 0 0 0 0 0 base map 100" /> 

</launch>